bokehjs/src/coffee/models/renderers/graph_renderer.coffee

```diff
@@ -66,11 +70,48 @@ export class GraphRenderer extends Renderer
       selector = @node_renderer.data_source.selection_manager.selector
       selector.update(indices, final, append)
       @node_renderer.data_source.selected = selector.indices
+
+      if @selection_mode == "linked"
+        node_indices = (@node_renderer.data_source.data.index[i] for i in indices["1d"].indices)
+        edge_source = @edge_renderer.data_source
+        edge_indices = []
+        for i in [0...edge_source.data.start.length]
+          if contains(node_indices, edge_source.data.start[i]) or contains(node_indices, edge_source.data.end[i])
+            edge_indices.push(i)
+
+        linked_index = create_hit_test_result()
+        for i in edge_indices
+          linked_index["2d"].indices[i] = [0] #currently only supports 2-element multilines, so this is all of it
+
+        edge_selector = @edge_renderer.data_source.selection_manager.selector
+        edge_selector.update(linked_index, final, append)
+        @edge_renderer.data_source.selected = edge_selector.indices
+        @edge_renderer.data_source.select.emit()
+
     else ## if mode=="inspect"
       inspector = @node_renderer.data_source.selection_manager.inspectors[@id]
       inspector.update(indices, true, false, true)
       @node_renderer.data_source.inspected = inspector.indices
 
+      if @inspection_mode == "linked"
+        node_indices = (@node_renderer.data_source.data.index[i] for i in indices["1d"].indices)
+        edge_source = @edge_renderer.data_source
+        edge_indices = []
+        for i in [0...edge_source.data.start.length]
+          if contains(node_indices, edge_source.data.start[i]) or contains(node_indices, edge_source.data.end[i])
+            edge_indices.push(i)
+
+        linked_index = create_hit_test_result()
+        for i in edge_indices
+          linked_index["2d"].indices[i] = [0] #currently only supports 2-element multilines, so this is all of it
+
+        # manually get or create inspector
+        @edge_renderer.data_source.selection_manager._set_inspector(@edge_renderer)
+        edge_inspector = @edge_renderer.data_source.selection_manager.inspectors[@edge_renderer.id]
+        edge_inspector.update(linked_index, final, append)
+        @edge_renderer.data_source.inspected = edge_inspector.indices
+        @edge_renderer.data_source.inspect.emit()
+
```

Comment:
making selection polices be models would also let the functionality be factored into chunks so that this does not have to be be a "huge switch" kind of function.